webpack/.eslintrc

```diff
@@ -26,6 +26,8 @@
       "specialLink": [ "to" ]
     }],
     "promise/prefer-await-to-then": "error",
-    "promise/prefer-await-to-callbacks": "error"
+    "promise/prefer-await-to-callbacks": "error",
+    "no-unused-vars": ["error", { "argsIgnorePattern": "^_" }],
+    "no-param-reassign": ["error", { "props": false }]
```

Comment:
this allows you to update a property of an object that is passed as an argument. e.g.:
```javascript
const fooList = [{ fruit: "apple" }, { fruit: "pear" }];
foo.map((fooList) => {
  fooList.vegetable = "celery"; // don't have to assign fooList to a new variable and mess with copying an object
}
```
The rule will error as normal otherwise "props" here means "property of object" (nothing to do with React), so it is only allowing that use case, which I think is fine, but open to suggestions if others are strictly opposed.

Reply:
it seems like there are other mechanisms in place to prevent this:

Even with the `Immutable` initial state removed, I still get this error when trying to update the state directly:

```
VM107 foreman-vendor.bundle-v4.3.0-development-94ef7563c5e7af8d70bc.js:383262 Uncaught TypeError: Cannot add property hi, object is not extensible
    at _default (VM120 katello.js:113675)
    at VM107 foreman-vendor.bundle-v4.3.0-development-94ef7563c5e7af8d70bc.js:383188
    at Array.forEach (<anonymous>)
    at assertReducerShape (VM107 foreman-vendor.bundle-v4.3.0-development-94ef7563c5e7af8d70bc.js:383186)
    at combineReducers (VM107 foreman-vendor.bundle-v4.3.0-development-94ef7563c5e7af8d70bc.js:383251)
    at Object.<anonymous> (VM120 katello.js:113826)
    at __webpack_require__ (VM109 vendor.js:713)
    at fn (VM109 vendor.js:118)
    at Object.<anonymous> (VM120 katello.js:113761)
    at __webpack_require__ (VM109 vendor.js:713)
```

This is with this change:
```diff
diff --git a/webpack/scenes/ContentViews/ContentViewReducer.js b/webpack/scenes/ContentViews/ContentViewReducer.js
index d3bbf92..a9e9d2b 100644
--- a/webpack/scenes/ContentViews/ContentViewReducer.js
+++ b/webpack/scenes/ContentViews/ContentViewReducer.js
@@ -8,11 +8,14 @@ import {
   CONTENT_VIEWS_FAILURE,
 } from './ContentViewConstants';
 
-const initialState = Immutable({
-  ...initialApiState,
-});
+//const initialState = Immutable({
+//  ...initialApiState,
+//});
+
+const initialState = {};
 
 export default (state = initialState, action) => {
+  state.hi = "hi"
```

Reply:
@jeremylenz  what is the best way to update the array of objects without updating the property of the argument like in the example above?

Reply:
I think it depends on the context and situation, but there are plenty of tools and options.  If you're dealing with Redux, this is exactly the purpose of reducers in the Redux store.  For components, there's also the React hook `useReducer()`, which is a non-Redux reducer but it's the same idea.  And also good old ES6 spread syntax :)

Reply:
come to think of it, `useReducer` might be great for the content view table

Reply:
> I think it depends on the context and situation, but there are plenty of tools and options. If you're dealing with Redux, this is exactly the purpose of reducers in the Redux store. For components, there's also the React hook `useReducer()`, which is a non-Redux reducer but it's the same idea. And also good old ES6 spread syntax :)

It is outside of Redux, for places like this 

https://github.com/Katello/katello/pull/8644/files#diff-10a686c727d0894d9c799f04ff780b1bR108-R111

Reply:
Oh yeah, for that I'd definitely say spread is best!